scripts/conflicts/diff.js

```diff
@@ -20,37 +26,90 @@ if (!server || argv.h || argv.help) {
 
 console.log(`Generating diffs for conflicts on ${server}`);
 
+
 const DB = new PouchDB(server);
 
-DB.query('medic-conflicts/conflicts')
+
+const writeToFile = function(path, content) {
+  const writeStream = fs.createWriteStream(path);
+  writeStream.write(JSON.stringify(content, null, 3));
+  writeStream.close();
+};
+
+DB.query('medic-conflicts/conflicts', {reduce:false})
   .then(conflicts => {
     console.log(`Found ${conflicts.rows.length} conflicts`);
 
+    const mainConflictFilesDirectory = './doc-conflicts';
     let ps = Promise.resolve();
+    if (!fs.existsSync(mainConflictFilesDirectory)){
+      try {
+        fs.mkdirSync(mainConflictFilesDirectory);
+      }catch(err){
+        console.log(`Error while creating main conflict folder: ${err}`);
+      }
+    }
     conflicts.rows.forEach(r => {
       ps = ps.then(() => {
-        // console.log(r);
-        if (r.key.length !== 1) {
-          console.log(`${r.id} has ${r.key.length} conflicts, we only want to deal with 1, skipping...`);
-        } else {
-          const docId = r.id;
-          const conflictRev = r.key[0];
-          return DB.get(docId, { open_revs: 'all' })
-            .then(results => {
-              // console.log(results);
-              // Make sure mine and theirs are the right way around
-              const mine = results.find(r => r.ok._rev !== conflictRev).ok;
-              const theirs = results.find(r => r.ok._rev === conflictRev).ok;
-
-              console.log('=====================');
-              console.log(r.id);
-              console.log('=====================');
-              console.log(jsondiff.diffString(mine, theirs));
-              console.log('\n\n');
-            });
-        }
+        const docId = r.id;
+        const arrayRevsIdThatShouldNotBeThere = r.key;
+        return DB.get(docId, { open_revs: 'all' })
+          .then(results => {
+
+            // Get the main_doc. Maindoc is the doc that isn't in the conflict list
+            const main_doc = results.filter(r => !arrayRevsIdThatShouldNotBeThere.includes(r.ok._rev))[0].ok; 
+
+            //Now we are going to put each doc, its conflicts and thier diff 
+            //in one file for manual inspection if required.
+            //The doc id might have characters that cannot be used to create folders.
```

Comment:
Which characters? 
If so, then it would be simpler to just replace those characters with acceptable ones, and apply the replacement for all ids, instead of having so many conditionals.

Reply:
Thank you for your feedback.

I have changed the replace regex to `conflictDirectoryName.replace(/[/\\?%*:|"<>]/g, '')` which should cover most if not all illegal characters.

Regarding multiple, if conditions.
The main goal was to make the folder name such that the folder name was part of the document ID in couch DB so that it is easily searchable if required.  I also wanted tasks , targets and contacts to be separated on the folders.

The patterns I saw in Target and task naming was that: 
Targets would require the **split to at the user colon** to have a unique searchable id like string.
`**target~2022-05~c7ddd59d-55f6-5c3c-832c-f13eb2869c83~org.couchdb.user**:bardiwas_chn4`

Whereas for **tasks we can't split at the colon** so only thing we can do is replace the colon with space.  
`task~org.couchdb.user**:**bardiwas_chn6~927796f2-d90b-44bc-a5aa-342326e5ddf7~pregnancy-home-visit-month-9~anc.pregnancy_home_visit.known_lmp~1652772300265`

Now the contact and targets docs are directly searchable, while for the task from we just need to replace the space after colon with space.

Reply:
I think the splitting and ID manipulation could look pretty arbitrary to someone who has not seen the code, and just using the document ID as is would be best.

Reply:
Document ID as is cannot be the folder name due to the ":" hence I had to split it.